cmd/k8sagent/initialize/config.go

```diff
@@ -130,13 +137,64 @@ func defaultConfig() agent.Config {
 	return &configFromEnv{}
 }
 
-type identityFunc func() identity
+type identityFunc func() (identity, error)
+
+func defaultIdentity() (identity, error) {
```

Comment:
Probably we can consider moving `func defaultIdentity(cloudType) (identity, error) {
` to `caas` package, and `identityFromECSMetadataService` -> `caas/kubernetes`, `identityFromK8sMetadata ` -> `caas/ecs` package.

To achieve this, we could inject `cloudType` in init container as an environment variable.

Reply:
Not quite sure if that would be a good idea. Granted, they are broker-specific but on the other hand they are meant to be invoked within a pod/task context as both implementations examine envvars (ok, the ecs one does an http get) to figure out the identity. As they are probably only going to be executed inside the k8sagent it makes sense to keep this logic encapsulated in there.

Reply:
ok, we can refactor if we need to later.